### PR TITLE
 #431 properties panels and conditions can specify class names 

### LINKED ISSUE #431: Properties panels and controls should have ability to specify a class name

### PR DESCRIPTION
**Pull request by caritaou**
closes #431 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/__tests__/common-properties/panels/control-test.js [hunk 1]
```diff
@@ -69,3 +69,24 @@ describe("control panel visible and enabled conditions work correctly", () => {
...
+		wrapper = renderedObject.wrapper;
+	});
+
+	afterEach(() => {
+		wrapper.unmount();
+	});
+
+	it("control panels should have custom classname defined", () => {
+		// Total of 27 classnames defined in text panels, but 1 is not in the DOM because of summary
+		expect(wrapper.find("[className$='-group-controls-class']")).to.have.length(26);
```
Comment (matthoward366) [inline canvas_modules/common-canvas/__tests__/common-properties/panels/control-test.js:86]: What is this testing above the test below?  I'm a little afraid that since this is so generic we won't really know what is specifically testing in the future.  Same goes for the other tests like this.

**Review by matthoward366 — COMMENTED** [review]
Changes look good.  In the test paramDefs it would be helpful if there was something in the UI that would tell us that a specific control should have a custom classname.  I'd have to look at each paramDef to know a specific control has a custom classname defined.

File: canvas_modules/common-canvas/__tests__/common-properties/panels/control-test.js (hunk 1 above)
Comment (caritaou) [reply canvas_modules/common-canvas/__tests__/common-properties/panels/control-test.js:86]: This is to verify that the number of classNames set in the paramDef matches the actual number of classNames found in the DOM. I agree this doesn't validate the actual panels itself, I will add additional tests to cover each type of panel

File: canvas_modules/common-canvas/src/common-properties/components/editor-form/editor-form.jsx [hunk 2]
```diff
@@ -314,11 +314,17 @@ class EditorForm extends React.Component {
...
-			return (<Subtabs key={"subtabs." + key} tabs={uiItem.tabs} controller={this.props.controller} rightFlyout={this.props.rightFlyout} genUIItem={this.genUIItem} />);
+			return (<Subtabs key={"subtabs." + key}
+				tabs={uiItem.tabs}
+				className={uiItem.className}
+				controller={this.props.controller}
+				rightFlyout={this.props.rightFlyout}
+				genUIItem={this.genUIItem}
+			/>);
 		case ("primaryTabs"):
 			return this.genPrimaryTabs(key, uiItem.tabs, inPropertyId, indexof);
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/components/editor-form/editor-form.jsx:325]: Is a user able to set a className on the top level groups?

File: canvas_modules/common-canvas/src/common-properties/form/EditorForm.js [hunk 3]
```diff
@@ -592,6 +607,7 @@ function _makeEditStyleSubPanel(structureDef, l10nProvider, structureMetadata) {
 	const panel = new ControlPanel(
 		structureDef.name,
 		PanelType.GENERAL,
+		"properties-editstyle-sub-panel",
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/form/EditorForm.js:610]: Why was this added?

Comment (caritaou) [reply canvas_modules/common-canvas/src/common-properties/form/EditorForm.js:610]: This className is added to all sub panel groups in tables, its not required and can be removed if we dont want to introduce another className

File: canvas_modules/common-canvas/src/common-properties/components/editor-form/editor-form.jsx (hunk 2 above)
Comment (caritaou) [reply canvas_modules/common-canvas/src/common-properties/components/editor-form/editor-form.jsx:325]: yes

**Review by matthoward366 — APPROVED** [review]
Changes look good.
